src/Common/src/Interop/Windows/kernel32/Interop.GetFileAttributesEx.cs

```diff
@@ -68,10 +67,8 @@ internal unsafe struct WIN32_FIND_DATA
             internal uint nFileSizeLow;
             internal uint dwReserved0;
             internal uint dwReserved1;
-            [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 260)]
-            internal string cFileName;
-            [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 14)]
-            internal string cAlternateFileName;
+            internal FixedString.Size260 cFileName;
+            internal FixedString.Size14 cAlternateFileName;
```

Comment:
> it introduces the possibility for others, e.g. every call to Value allocating a new string.

Well, you introduce that problem with any solution that doesn't allow the marshaller to convert it to a string for you.

> I don't see how these structs help to eliminate mistakes more than helpers do

Primarily by not having the ability to accidentally set the wrong size and encapsulating unsafe access. Span<char> does address part of this.

> If you would like to have something that looks pretty and that is safe as much as possible, you should use Span<T>.

I'm ok with trying that as a middle ground. Any suggestions on what the helper class should be named? Should I use Span<char> extension methods?

> We may want to add automatic conversions from fixed buffer field to Span/ReadOnlySpan to C# - at which point these small getters can go away.

Sounds like a great thing to me.